index.js

```diff
@@ -73,35 +74,79 @@ async function run() {
         core.setFailed('"arch" may only be x86 or x86_64');
       }
 
-      const installDir =
-        process.env.GSTREAMER_INSTALL_DIR ?? path.join(isSelfHosted() ? 'C:' : 'D:', 'gstreamer');
-
-      const installers = [
-        `gstreamer-1.0-msvc-${arch}-${version}.msi`,
-        `gstreamer-1.0-devel-msvc-${arch}-${version}.msi`,
-      ];
-
-      for (const installer of installers) {
-        const url = `${baseUrl}/windows/${version}/msvc/${installer}`;
-
-        core.info(`Downloading: ${url}`);
-        const installerPath = await tc.downloadTool(url, installer);
-
-        if (installerPath) {
-          await exec.exec('msiexec', [
-            '/passive',
-            `INSTALLDIR=${installDir}`,
-            'ADDLOCAL=ALL',
-            '/i',
-            installerPath,
-          ]);
-          await io.rmRF(installerPath);
-        } else {
-          core.setFailed(`Failed to download ${url}`);
+      if (buildSource) {
+        const installDir =
+          process.env.GSTREAMER_INSTALL_DIR ?? path.join(isSelfHosted() ? 'C:' : 'D:', `gstreamer\\1.0\\msvc_${arch}`);
+
+        const sourceDir = path.join(isSelfHosted() ? 'C:' : 'D:', 'gstreamer_source');
+
+        core.info("Cloning gstreamer's git repository...");
+        await exec.exec('git', ['config', '--global', 'http.postBuffer', '524288000']);
+        await exec.exec('git', [
+          'clone',
+          '--progress',
+          '--verbose',
+          '--depth',
+          '1',
+          '--branch',
+          version,
+          gitUrl,
+          sourceDir,
+        ]);
+
+        const sourceTarget = { cwd: `${sourceDir}` };
+        const buildArguments = core.getInput('gstreamerOptions').split(',');
```

Comment:
These should be used in Linux too. Move this out to the global scope (probably renamed to `userBuildArgs`) and use these in both Windows:
```js
const windowsBuildArgs = [
  '--buildtype=debugoptimized'
];
const buildArgs = combineBuildArgs(windowsBuildArgs, userBuildArgs);

await exec.exec(
  'meson',
  [
    'setup', 
    '--vsenv',
    `--prefix=${installDir}`,
    ...buildArgs,
    'builddir',
  ],
  sourceTarget
);
```
and Linux:
```js
const linuxBuildArgs = [
  '-Dges=disabled',
  '-Dtests=disabled',
  '-Dexamples=disabled',
  '-Dgst-examples=disabled',
  '-Ddoc=disabled',
  '-Dgtk_doc=disabled',
  '-Dgpl=enabled',
];
const buildArgs = combineBuildArgs(windowsBuildArgs, userBuildArgs);

await exec.exec(
  'meson',
  [
    'setup',
    `--prefix=${prefix}`,
    ...buildArgs,
    'builddir'
  ],
  opt
);
```

Where `combineBuildArgs` overrides the default build arguments with those specified by the user:
```js
const combineBuildArgs = (systemArgs, userArgs) =>
  systemArgs
    .filter((systemArg) => !userArgs.some((userArg) => userArg.startsWith(systemArg.split('=')[0])))
    .concat(userArgs);
```